OneCommunityGlobal/HighestGoodNetworkApp
# Snehal fix user id dashboard

# Description
Fix user ID workaround bug for dashboard display:
Bug: The Add Time Entry button on the Dashboard displays the wrong user name after viewing another person’s profile.
The Badges section also shows wrong profile data after viewing another person’s profile.

## Related PRS (if any):
To test this frontend PR you need to checkout the development branch in backend.

## Main changes explained:
Changes in Dashboard.jsx file 

## How to test:
Test 1:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as Owner user -> Login into Owner's  Dashboard
5. Leaderboard -> click on red/green circular button of another user -> This will display <another user> viewing dashboard. 
6. Popup to view another user dashboard -> Click in "OK"
7. Check the name on summary bar, Badges and "Add time .." button name.
8. View yellow bar "You are currently viewing header for .. " Click on "X" to exit form the another user viewing dashboard page.
9. Popup to exit the another user view dashboard -> "OK"
10. This will send to current user viewing page. Check the summary name, badges and Yellow button "Add time" name.

Test 2: 
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as Owner user -> Login into Owner's  Dashboard
5. Click on any user from leaders board -> Land to userprofile of selected user (http://localhost:3000/userprofile/<another_userid>)
6. Change the URL to ( http://localhost:3000/dashboard/<another_userid> )
7. Check the name on summary bar, Badges and "Add time .." button name. (Should be another user details)

## Screenshots or videos of changes:
Test 1: 
https://github.com/user-attachments/assets/9c84e392-211f-4615-adce-3669c0faf89e

Test 2:
https://github.com/user-attachments/assets/32f843ca-5690-4541-bc63-16a2e41742ac
